proposals/stdlib/locale-agnostic-string-conversions.md

```diff
@@ -0,0 +1,159 @@
+ */
+@Deprecated("Please use locale-insensitive alternative `decapitalizeFirst()`", ReplaceWith("decapitalizeFirst()"))
+public expect fun String.decapitalize(): String
+
+/**
+ * Returns a copy of this string having its first letter lowercased using Unicode mapping rules of the invariant locale,
+ * or the original string if it's empty or already starts with a lower case letter.
+ */
+public expect fun String.decapitalizeFirst(): String
+```
+
+The new namings will also affect following functions:
+
+* Rename `String.toLowerCase(locale: Locale): String` to `String.lowercase(locale: Locale): String`
+* Rename `String.toUpperCase(locale: Locale): String` to `String.uppercase(locale: Locale): String`
+* Rename `String.capitalize(locale: Locale): String` to `String.capitalizeFirst(locale: Locale): String`
+* Rename `String.decapitalize(locale: Locale): String` to `String.decapitalizeFirst(locale: Locale): String`
+* Rename `Char.toLowerCase(): Char` to `Char.lowercase(): Char`
+* Rename `Char.toUpperCase(): Char` to `Char.uppercase(): Char`
+* Rename `Char.toTitleCase(): Char` to `Char.titlecase(): Char`
+
+```kotlin
+@Deprecated("This function has been renamed to `lowercase`", ReplaceWith("lowercase(locale)"))
+public fun String.toLowerCase(locale: java.util.Locale): String = lowercase(locale)
+
+public fun String.lowercase(locale: java.util.Locale): String = (this as java.lang.String).toLowerCase(locale)
+
+@Deprecated("This function has been renamed to `uppercase`", ReplaceWith("uppercase(locale)"))
+public fun String.toUpperCase(locale: java.util.Locale): String = uppercase(locale)
+
+public fun String.uppercase(locale: java.util.Locale): String = (this as java.lang.String).toUpperCase(locale)
+
+@Deprecated("This function has been renamed to `lowercase()`", ReplaceWith("lowercase()"))
+public fun Char.toLowerCase(): Char = lowercase()
+
+public expect fun Char.lowercase(): Char
+
+@Deprecated("This function has been renamed to `uppercase()`", ReplaceWith("uppercase()"))
+public fun Char.toUpperCase(): Char = uppercase()
+
+public expect fun Char.uppercase(): Char
+
+@Deprecated("This function has been renamed to `titlecase()`", ReplaceWith("titlecase()"))
+public fun Char.toTitleCase(): Char = titlecase()
+
+public fun Char.titlecase(): Char = Character.toTitleCase(this)
+```
+
+By renaming the functions we get rid of Java legacy names. 
+In Kotlin functions starting with preposition `to` convert receiver to instance of another type.
+
+## Dependencies
+
+No additional dependencies are needed.
+
+## Placement
+
+- module `kotlin-stdlib`
+- package `kotlin.text`
+
+## Reference implementation
+
+The reference implementation is provided in the pull request [PR #3780](https://github.com/JetBrains/kotlin/pull/3780).
+
+## Naming
+
+Alternative naming suggestions are welcome.
+
+## Compatibility impact
+
+The deprecated functions will go through standard deprecation cycle.
```

Comment:
Do we have a reference to Kotlin standard deprecation cycle? Maybe @ilya-g knows?

Reply:
There is a deprecation cycle template in Language Committee Guidelines: https://kotlinlang.org/foundation/language-committee-guidelines.html#incompatible-changes-to-the-libraries
However, strengthening deprecation level for each individual function is usually discussed before a major Kotlin release.

Reply:
The standard deprecation cycle implies advancing the deprecation severity with each major release until the function is hidden or completely removed. But since we do not plan to reuse the old names, we can stop moving through this cycle at the deprecation severity `ERROR` or even `WARNING`.